web/ui/react-app/src/pages/graph/GraphHelpers.ts

```diff
@@ -195,3 +230,19 @@ export const parseValue = (value: string) => {
   // can't be graphed, so show them as gaps (null).
   return isNaN(val) ? null : val;
 };
+
+const exemplarSymbol = (ctx: CanvasRenderingContext2D, x: number, y: number) => {
```

Comment:
Btw. I just played around and find the default points (when omitting the custom symbol function) more beautiful, plus they have a nice hover effect :) Did you mainly switch to a custom method for size reasons (which should be controllable by the `radius` property)?

Reply:
Yes, it was mainly for size reasons. I don't believe you can control the radius per-point, only for all of them (series and exemplars), so the custom symbol was the only option. The square also creates distinction between it and a regular series point.

Reply:
Hmm... so I can definitely increase the radius of only the exemplar circles, and normal series are never drawn as circles, only as line segments, so there should at least be a distinction already.

```typescript
points: { show: true, radius: 4 },
```

Gives me:

https://user-images.githubusercontent.com/538008/119251944-83cc3f00-bba9-11eb-97b6-6d496b962814.mp4

The nice thing about this option is that it works with the existing hover plugin and thus meets the user expectation of indicating which exemplar (or other point) you are currently hovering over.

On the custom drawing alternative front I also experimented with making the exemplars a bit more pretty (IMO), like this:

```typescript
const exemplarSymbol = (ctx: CanvasRenderingContext2D, x: number, y: number) => {
  // Center the symbol on the point.
  y = y - 3.5;

  // Correct if the symbol is overflowing off the grid.
  if (x > ctx.canvas.clientWidth - 59) {
    x = ctx.canvas.clientWidth - 59;
  }
  if (y > ctx.canvas.clientHeight - 37) {
    y = ctx.canvas.clientHeight - 37;
  }

  ctx.translate(x, y);
  ctx.rotate(Math.PI / 4);
  ctx.translate(-x, -y);

  ctx.fillStyle = '#92bce1';
  ctx.fillRect(x, y, 7, 7);

  ctx.strokeStyle = '#0275d8';
  ctx.lineWidth = 1;
  ctx.strokeRect(x, y, 7, 7);
}
```

That gives us:

https://user-images.githubusercontent.com/538008/119252021-06ed9500-bbaa-11eb-893f-bdec17a11e46.mp4

But the downside is that hovering doesn't show a highlight. I guess we could build our own specific hover functionality, like the hover plugin (http://www.flotcharts.org/flot/source/jquery.flot.hover.js) does it for circles.

I think either of those solutions is fine as a start and we shouldn't block the PR merge on more though.

Reply:
Weird, I think I tried changing the radius and it messed up the lines, but cool that it works! I think that your custom symbols fit in a lot better and look really good, so I'll go ahead and add that in. Hover can definitely come later.